### PR TITLE
ENH: Standalone CLI now estimates fieldmaps

### PR DESCRIPTION
**Pull request by oesteban**
The previous behavior (just "finding" estimators) is maintained with the argument ``--dry-run``.

### CODE REVIEW

**Comment by oesteban** [comment]
Well, at df93c7b, this is surprisingly executing the estimation of GRE fieldmaps. EPI fieldmaps failed in the robust average, so I'll need to check what went wrong.

File: .github/workflows/docs-build-pr.yml
Comment (effigies) [inline .github/workflows/docs-build-pr.yml]: What's the need for an older version? 3 was specifically intended to avoid pinning. The default Python action marches along at a reasonably slow pace, so we shouldn't need to worry about 3.12 becoming default before the ecosystem is ready, but by now it should be safe to build on 3.11.

File: docs/_static/config-example.toml
Comment (effigies) [inline docs/_static/config-example.toml:33]: This is expected to be a list?

File: sdcflows/cli/main.py
Comment (effigies) [inline sdcflows/cli/main.py:122]: Is it needed to separate this, if you've already initialized the process pool?

File: sdcflows/cli/main.py
Comment (effigies) [inline sdcflows/cli/main.py]: Unused?

File: .github/workflows/docs-build-pr.yml
Comment (oesteban) [reply .github/workflows/docs-build-pr.yml]: Not sure why I did this -- rolling back

```suggestion
        python-version: 3
```

File: .github/workflows/docs-build-update.yml
Comment (oesteban) [inline .github/workflows/docs-build-update.yml]:
```suggestion
        python-version: 3
```

File: sdcflows/cli/parser.py
Comment (effigies) [inline sdcflows/cli/parser.py]: I would be inclined not to have synonyms when they aren't required for backwards compatibility.

```suggestion
        "--participant-label",
```

File: sdcflows/cli/parser.py
Comment (effigies) [inline sdcflows/cli/parser.py]: For consistency

```suggestion
        "--session-label",
```

File: sdcflows/cli/parser.py
Comment (effigies) [inline sdcflows/cli/parser.py]: Again, I would pick one and go with it. I don't care which.

File: sdcflows/cli/parser.py
Comment (effigies) [inline sdcflows/cli/parser.py]:
```suggestion
```

File: sdcflows/cli/parser.py
Comment (effigies) [inline sdcflows/cli/parser.py]:
```suggestion
```

File: sdcflows/cli/parser.py
Comment (effigies) [inline sdcflows/cli/parser.py]:
```suggestion
        help="Enable changes to processing to aid in debugging",
```

File: sdcflows/cli/parser.py
Comment (effigies) [inline sdcflows/cli/parser.py]: IMO it should be considered a first-class option. It's legitimate to disable it, but I think we should default to True.

```suggestion
        "--no-fmapless",
        action="store_false",
        dest="fieldmapless",
        default=False,
```

File: sdcflows/cli/parser.py
Comment (effigies) [inline sdcflows/cli/parser.py]: Spacing is off. Adding a semantic line break.

```suggestion
        help="Opt-out of sending tracking information of this run to the NiPreps developers. "
        "This information helps to improve SDCFlows and provides an indicator of "
        "real world usage for obtaining funding.",
```

File: sdcflows/cli/parser.py
Comment (effigies) [inline sdcflows/cli/parser.py:120]: What is a group-level analysis?

File: docs/_static/config-example.toml
Comment (oesteban) [reply docs/_static/config-example.toml:33]: yup (for laziness). It *could* be `[ "participant", "group" ]`, however group currently doesn't have an implementation (perhaps generating a group report?)

File: sdcflows/cli/main.py
Comment (oesteban) [reply sdcflows/cli/main.py:122]: We could send it to the pool, that'd be fine. However, it's working this way in MRIQC.

File: sdcflows/cli/main.py
Comment (oesteban) [reply sdcflows/cli/main.py]:
```suggestion
```

File: sdcflows/cli/parser.py
Comment (oesteban) [reply sdcflows/cli/parser.py:120]: Right now, it's not a thing.

File: sdcflows/cli/parser.py
Comment (oesteban) [inline sdcflows/cli/parser.py]:
```suggestion
```

File: sdcflows/config.py
Comment (oesteban) [inline sdcflows/config.py]:
```suggestion
    session_label = None
```

File: sdcflows/cli/main.py
Comment (effigies) [reply sdcflows/cli/main.py:122]: My point was the pool is initialized, so isn't it safe to do this in the main process at this point? These managers tend to make debugging trickier, so if we can avoid it, I'd rather.

**Review by effigies — APPROVED** [review]
Go ahead and merge this when you're ready. It would be nice if we could get away from the `Manager()` indirection for building workflows eventually, but I don't want to hold this up on that.

File: sdcflows/cli/main.py
Comment (oesteban) [reply sdcflows/cli/main.py:122]: Oh, yes, in theory we could use the main process without problems, indeed.

Comment (oesteban) [reply sdcflows/cli/main.py:122]: Okay, I was removing this context to build the workflow in the main thread, and it could be interesting to keep it this way, as the workflow is created while OMP_NUM_THREADS is set to 1. This saves great numpy buildup. 

We could still move it to the main process, but then it would make sense to make sure numpy is imported with OMP_NUM_THREADS = 1 (and that is then overridden in  subprocesses).

**Pull request merged**
